python/lsst/skymap/patchInfo.py

```diff
@@ -98,30 +143,218 @@ def getOuterBBox(self):
+        ------
+        IndexError
+            If index is out of range.
+        """
+        if self._numCells[0] == 0 or self._numCells[1] == 0:
+            raise IndexError("Patch does not contain cells.")
+        if isinstance(index, Index2D):
+            _index = index
+        else:
+            if isinstance(index, numbers.Number):
+                _index = self.getCellIndexPair(index)
+            else:
+                _index = Index2D(*index)
+        if (not 0 <= _index.x < self._numCells[0]) \
+                or (not 0 <= _index.y < self._numCells[1]):
+            raise IndexError("Cell index %s is not in range [0-%d, 0-%d]" %
+                             (_index, self._numCells[0] - 1, self._numCells[1] - 1))
+        # We offset the index by numCellsInPatchBorder because the cells
+        # start outside the inner dimensions.
+        # The cells are defined relative to the patch bounding box (within the tract).
+        patchInnerBBox = self.getInnerBBox()
+        innerMin = Point2I(*[(_index[i] - self._numCellsInPatchBorder)*self._cellInnerDimensions[i]
+                             + patchInnerBBox.getBegin()[i]
+                             for i in range(2)])
+
+        innerBBox = Box2I(innerMin, self._cellInnerDimensions)
+        outerBBox = Box2I(innerBBox)
+        outerBBox.grow(self._cellBorder)
+
+        return CellInfo(
+            index=_index,
+            innerBBox=innerBBox,
+            outerBBox=outerBBox,
+            sequentialIndex=self.getSequentialCellIndexFromPair(_index),
+            tractWcs=self._wcs
+        )
+
+    def getCellInnerDimensions(self):
+        """Get dimensions of inner region of the cells (all are the same)
+        """
+        return self._cellInnerDimensions
+
+    cell_inner_dimensions = property(getCellInnerDimensions)
+
+    def getSequentialCellIndex(self, cellInfo):
+        """Return a single integer that uniquely identifies
+        the given cell within this patch.
+
+        Parameters
+        ----------
+        cellInfo : `lsst.skymap.CellInfo`
+
+        Returns
+        -------
+        sequentialIndex : `int`
+
+        Raises
+        ------
+        IndexError
+            If index is out of range.
+        """
+        index = cellInfo.getIndex()
+        return self.getSequentialCellIndexFromPair(index)
+
+    def getSequentialCellIndexFromPair(self, index):
+        """Return a single integer that uniquely identifies
+        the given cell within this patch.
+
+        Parameters
+        ----------
+        index : `lsst.skymap.Index2D`
+
+        Returns
+        -------
+        sequentialIndex : `int`
+
+        Raises
+        ------
+        IndexError
+            If index is out of range.
+        """
+        if isinstance(index, Index2D):
+            _index = index
+        else:
+            if not isinstance(index, Iterable):
+                raise ValueError("Input index is not an iterable.")
+            if len(index) != 2:
```

Comment:
The `Index2D` constructor (or more accurately the `NamedTuple` constructor) takes care of this internally that I don't think this check needs to be in here.